### PR TITLE
[apr-util] update dependencies, remove openssl fix

### PR DESCRIPTION
**Pull request by mathbunnyru**
Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SpaceIm — DISMISSED** [review]

**Comment by mathbunnyru** [comment]
https://github.com/conan-io/conan-center-index/pull/4963
I made a fix to disable apr on xcode 12, because it is not buildable.

An issue
https://github.com/conan-io/conan-center-index/issues/3008

Unsuccessful attempt to fix the problem:
https://github.com/conan-io/conan-center-index/pull/3014

**Review by prince-chrismc — APPROVED** [review]

**Review by theirix — APPROVED** [review]
Good job, 4597 really helps

File: recipes/apr-util/all/conanfile.py [hunk 1]
```diff
@@ -95,16 +98,20 @@ def requirements(self):
...
-            self.requires("libpq/11.5")
+            self.requires("libpq/13.2")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
 
+    def validate(self):
+        if self.options.shared != self.options["apr"].shared:
+            raise ConanInvalidConfiguration("apr-util must be built with same shared option as apr")
```
Comment (SpaceIm) [inline recipes/apr-util/all/conanfile.py:113]: @madebr, this is why forcing upstream option in configure() is useful, here shared apr-util is not tested.

**Review by SpaceIm — APPROVED** [review]

File: recipes/apr-util/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/apr-util/all/conanfile.py:113]: Yes, but that can override user's options if it is built with `-o apr:shared=False -o apr-util=True`.
You can override the option if you absolutely know that the option is not passed, but I think this behavior can get complex and hard to understand fast.

Comment (SpaceIm) [reply recipes/apr-util/all/conanfile.py:113]: Yes, it's an inconsistency in conan side. Overriding rules are not the same depending if upstream options are set through command lines or a conanfile.py.

**Comment by mathbunnyru** [comment]
> Good job, 4597 really helps

Yep, I think we reduced almost all of the patching required to make openssl work.
There are other libraries as well with similar problems, though.

**Review by SSE4 — APPROVED** [review]

**Pull request merged**